### PR TITLE
use EnvironmentPlugin instead of DefinePlugin

### PR DESCRIPTION
**Pull request by mmvsha**
This PR introduces use of EnvironmentPlugin instead of DefinePlugin for managing environment variables. Also fixes default values for env variables inside production config

### CODE REVIEW

File: webpack.dev.config.js [hunk 1]
```diff
@@ -5,121 +5,118 @@ const path = require('path')
...
 
-const nodeEnv = process.env.NODE_ENV || 'development'
 const version = process.env.BUILD_VERSION || pkg.version
 const build = process.env.BUILD_NUMBER || 'SNAPSHOT'
-const intercomId = process.env.INTERCOM_ID
 
 const config = require('./src/config.json')
 
-const whitelist = config.productionWhitelist
+const whitelist = config.developmentWhitelist
```
Comment (apanizo) [inline webpack.dev.config.js]: You sure? @denisgranha told me I had to use the `productionWhitelist`

Comment (mmvsha) [reply webpack.dev.config.js]: @apanizo why use production whitelist for a development build?

Comment (apanizo) [reply webpack.dev.config.js]: Do not know...., ask him in slack, and if he gives you the ok feel free to merge it.

Comment (denisgranha) [reply webpack.dev.config.js]: Don't remember the time when we discussed this, but If I said that, I was wrong @apanizo we should use dev whitelist for development

**Pull request merged**